pkg/webhook/validate.go

```diff
@@ -189,98 +185,103 @@ func dynamicValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePo
 	for _, node := range nodeList.Items {
 		if cr.Selected(&node) {
 			nodesSelected = true
-			for _, ns := range nsList.Items {
-				if ns.GetName() == node.GetName() {
-					if ok, err := validatePolicyForNodeState(cr, &ns, &node); err != nil || !ok {
-						return false, err
-					}
-				}
-			}
-			// validate current policy against policies in API (may not be converted to SriovNetworkNodeState yet)
-			for _, np := range npList.Items {
-				if np.GetName() != cr.GetName() && np.Selected(&node) {
-					if ok, err := validatePolicyForNodePolicy(cr, &np); err != nil || !ok {
-						return false, err
-					}
-				}
+			err = validatePolicyForNodeStateAndPolicy(nsList, npList, &node, cr)
+			if err != nil {
+				return false, err
 			}
 		}
 	}
 
 	if !nodesSelected {
 		return false, fmt.Errorf("no matched node is selected by the nodeSelector in CR %s", cr.GetName())
 	}
-	if !interfaceSelected {
```

Comment:
Cleanup should only mean refactoring. You are changing the behaviour of the code here. We want this error to be printed when no interface is found.

Reply:
Okay, so I reverted this to what it was before. You can find the error when no interface is found to be in the other PR.